Add Document controller documentation (#32)
kuzzleio/sdk-csharp: doc/1/controllers/document/m-update/index.md

---
code: true
type: page
title: MUpdateAsync
description: Updates documents.
---

# MUpdateAsync

Updates multiple documents.

> Returns a partial error (error code 206) if one or more documents can not be updated.

Reviewer: Never write `can not` (unless [in a few particular cases](https://grammarist.com/usage/cannot-or-can-not/)):

```suggestion
Returns a partial error (error code 206) if one or more documents cannot be updated.
```

Conflicts may occur if the same document gets updated multiple times within a short timespan in a database cluster.

You can set the `retryOnConflict` optional argument (with a retry count), to tell Kuzzle to retry the failing updates the specified amount of times before rejecting the request with an error.

## Arguments

```csharp
public async Task<JArray> MUpdateAsync(
  string index, 
  string collection, 
  JArray documents, 
  bool waitForRefresh = false, 
  int retryOnConflict = 0);

```

<br/>

| Argument     | Type                                 | Description                                      |
| ------------ | ------------------------------------ | ------------------------------------------------ |
| `index`      | <pre>string</pre>        | Index name                                       |
| `collection` | <pre>string</pre>        | Collection name                                  |

> | `documents`  | <pre>JObject</pre>        | JObject representing the documents to update |

Reviewer 2: ```suggestion
| `documents`  | <pre>JArray</pre>        | JArray of documents to update |
```
